grabbit/src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -88,10 +90,51 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
             }
         }
 
-        final List<JcrProperty> properties = jcrNode.properties.toList()
-        Node.Builder nodeBuilder = Node.newBuilder()
+        // It checks if the node parent has required primary type as "nt:hierarchyNode"
+        // and then skip it as the node has already been processed by parent node
+        if(isRequiredHierarchyChildNode(jcrNode)) {
```

Comment:
Wondering if we should encapsulate this logic within JCRNodeDecorator, instead of having JcrNodesProcessor inspect node instances.  Similar to ProtoNodeDecorator, JCRNodeDecorator should "know" how to write itself to proto format, as well as perform any necessary inspection on it's underlying inner node.    

It would spring board enthusiasm to get toProperty()'s ugly switch statement dissolved in another effort, in a similar manner of empowering our objects.